src/_RelatedViewWidgetListMixin.js

```diff
@@ -0,0 +1,258 @@
+                    for (i = 0; i < this.relatedViews.length; i++) {
+                        if (this.relatedViews[i].enabled) {
+                            relatedViewManager = this.getRelatedViewManager(this.relatedViews[i]);
+                            if (relatedViewManager) {
+                                if (!entry.$key) {
+                                    entry.$key = this.store.getIdentity(entry);
+                                }
+                                relatedViewManager.addView(entry, rowNode, this);
+                            }
+                        }
+                    }
+                }
+                catch (error) {
+                    console.log('Error processing related views:' + error);
+
+                }
+            }
+        },
+        /**
+         *  Destroys all of the related view widgets, that was added.
+         */
+        destroyRelatedViewWidgets: function() {
+            var relatedViewId;
+            if (this.relatedViewManagers) {
+                for (relatedViewId in this.relatedViewManagers) {
+                    if (this.relatedViewManagers.hasOwnProperty(relatedViewId)) {
+                        this.relatedViewManagers[relatedViewId].destroyViews();
+                    }
+                }
+            }
+        },
+        /**
+         * Extends dijit Widget to destroy the search widget before destroying the view.
+         */
+        destroy: function() {
+            this.destroyRelatedViewWidgets();
+            this.inherited(arguments);
+        },
+        clear: function(all) {
+            this.inherited(arguments);
+            this.destroyRelatedViewWidgets();
+        },
+        /**
+         * Returns a rendered html snap shot of the entry.
+         */
+        getContextSnapShot: function(options) {
+            var snapShot, entry = this.entries[options.key];
+            if (entry) {
+                snapShot = this.itemTemplate.apply(entry, this);
+            }
+            return snapShot;
+        },
+        destroyRelatedView: function(relatedView) {
+            var relatedViewManager;
+            if (relatedView) {
+                relatedViewManager = this.getRelatedViewManager(relatedView);
+                if (relatedViewManager) {
+                    relatedViewManager.destroyViews();
+                }
+            }
+        },
+        invokeRelatedViewAction: function(action, selection, additionalOptions) {
+            var relatedView,
+                relatedViewManager,
+                relatedNode,
+                entry,
+                addView,
+                selectedRow,
+                selectedItems,
+                scrollerNode,
+                key;
+
+            addView = true;
+            relatedView = action['relatedView'];
+            if (!relatedView.hasOwnProperty('enabled')) {
+                relatedView.enabled = true;
+            }
+
+            if (relatedView && relatedView.enabled) {
```

Comment:
I would clean up these deep nested ifs by inverting them and do a simple return to bail. You are also checking if relatedView is truthy in this if statement, but above this on line 180 you are calling hasOwnProperty on it without doing a check if it is null or undefined.